### ISSUE TITLE
Broblem with compiling perl5

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#495]
Migrated from [rt.perl.org#1337](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=1337) (status was 'resolved')

Searchable as RT1337$

**Comment by p5pRT** [comment]
### From Richard.Meijn@nl.abnamro.com

This is my configuration

root@&#8203;tws\-83&#8203;:/tmp/perl5\.005\_03 \# \./myconfig
Summary of my perl5 \(5\.0 patchlevel 5 subversion 3\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=hpux\, osvers=10\.20\, archname=PA\-RISC1\.1
&nbsp;    uname='hp\-ux tws\-83 b\.10\.20 a 9000715 \-1 two\-user license '
&nbsp;    hint=previous\, useposix=true\, d\_sigaction=define
&nbsp;    usethreads=undef useperlio=undef d\_sfio=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='/opt/gcc/bin/gcc'\, optimize=' '\, gccversion=2\.8\.1
&nbsp;    cppflags='\-D\_HPUX\_SOURCE'
&nbsp;    ccflags ='\-D\_HPUX\_SOURCE'
&nbsp;    stdchar='unsigned char'\, d\_stdstdio=define\, usevfork=false
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=8
&nbsp;    alignbytes=8\, usemymalloc=y\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='ld'\, ldflags =' \-L/usr/local/lib 
\-L/opt/gcc/lib/gcc\-lib/hppa1\.1\-hp\-hpux10\.20/2\.8\.1/include 
\-L/usr/include/sys'
&nbsp;    libpth=/opt/gcc/lib/gcc\-lib/hppa1\.1\-hp\-hpux10\.20/2\.8\.1/include 
/usr/lib /usr/lib/pa1\.1 /usr/local/lib /lib/pa1\.1 /lib /usr/ccs/lib 
/usr/include /usr/include/sys /opt/gcc/include/g\+\+
&nbsp;    libs=\-lnsl\_s \-lndbm \-ldld \-lm \-lc \-lndir \-lcrypt
&nbsp;    libc=/usr/lib/libc\.sl\, so=sl\, useshrplib=false\, libperl=libperl\.a
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_hpux\.xs\, dlext=sl\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E 
\-Wl\,\-B\,deferred '
&nbsp;    cccdlflags='\-fpic'\, lddlflags='\-b \-L/usr/local/lib 
\-L/opt/gcc/lib/gcc\-lib/hppa1\.1\-hp\-hpux10\.20/2\.8\.1/include 
\-L/usr/include/sys'


If I try to compile PERL5 I get the following problem\.

root@&#8203;tws\-83&#8203;:/tmp/perl5\.005\_03 \# make
&nbsp;        \`sh  cflags libperl\.a mg\.o\`  mg\.c
&nbsp;          CCCMD =  /opt/gcc/bin/gcc \-DPERL\_CORE \-c \-D\_HPUX\_SOURCE     
In file included from 
/opt/gcc/lib/gcc\-lib/hppa1\.1\-hp\-hpux10\.20/2\.8\.1/include/stdlib\.h&#8203;:231\,
&nbsp;                 from perl\.h&#8203;:373\,
&nbsp;                 from mg\.c&#8203;:16&#8203;:
/opt/gcc/lib/gcc\-lib/hppa1\.1\-hp\-hpux10\.20/2\.8\.1/include/sys/wait\.h&#8203;:102&#8203;: 
warning&#8203;: \`struct rusage' declared inside parameter list
/opt/gcc/lib/gcc\-lib/hppa1\.1\-hp\-hpux10\.20/2\.8\.1/include/sys/wait\.h&#8203;:102&#8203;: 
warning&#8203;: its scope is only this definition or declaration\,
/opt/gcc/lib/gcc\-lib/hppa1\.1\-hp\-hpux10\.20/2\.8\.1/include/sys/wait\.h&#8203;:102&#8203;: 
warning&#8203;: which is probably not what you want\.
mg\.c&#8203;: In function \`Perl\_magic\_get'&#8203;:
mg\.c&#8203;:619&#8203;: \`none' undeclared \(first use in this function\)
mg\.c&#8203;:619&#8203;: \(Each undeclared identifier is reported only once
mg\.c&#8203;:619&#8203;: for each function it appears in\.\)
mg\.c&#8203;:619&#8203;: parse error before \`gary'
mg\.c&#8203;:620&#8203;: \`gary' undeclared \(first use in this function\)
mg\.c&#8203;: In function \`Perl\_magic\_set'&#8203;:
mg\.c&#8203;:1755&#8203;: \`none' undeclared \(first use in this function\)
mg\.c&#8203;:1755&#8203;: parse error before \`gary'
mg\.c&#8203;:1768&#8203;: \`gary' undeclared \(first use in this function\)
\*\*\* Error exit code 1


How could I solve the problem\.

If you need more information please let me know




Kind regards\, 

Richard Meijn
Afdeling     : TTF
Tel\.           : \+31 \(0\)20\-3830845               Fax\.      : \+31 
\(0\)20\-6289682
PAC          : AA1120                              Lokatie  : 
Foppingadreef 22
E\-mail       : richard\.meijn@&#8203;nl\.abnamro\.com
